scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,377 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+This file contains all the code that has to do with editing the remote repo template
+"""
+import argparse
+import pathlib
+import sys
+import json
+import logging
+import hashlib
+import shutil
+from packaging.version import Version, InvalidVersion
+from packaging.specifiers import SpecifierSet
+from o3de import manifest, utils, validation
+
+logger = logging.getLogger('o3de.repo_properties')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+def merge_json_data(json_path: pathlib.Path, json_data: dict) -> dict:
+    """
+    Merges the json data with the existing json data if it exists.
+    If the json data does not exist, it is created.
+    The json data is saved to the json path.
+    Returns the merged json data.
+    
+    Args:
+        json_path (pathlib.Path): The path to the json file.
+        json_data (dict): The json data to merge with the existing json data.
+    
+    Returns:
+        dict: The merged json data.
+    
+    Raises:
+        FileNotFoundError: If the json path does not exist.
+        ValueError: If the json data is not a dict.
+    """
+    merged_json_data = {}
+    if json_path.exists():
+        existing_json_data = {}
+        with open(json_path, 'r') as f:
+            existing_json_data = json.load(f)
+
+        merged_json_data = existing_json_data | json_data
```

Comment:
We should check here if the data is the same and not modify the file if we don't have to which causes the file modification time to change and might make the file look modified to source control.
```suggestion
        with open(json_path, 'r') as f:
            existing_json_data = json.load(f)

        merged_json_data = existing_json_data | json_data
        
        if merged_json_data == existing_json_data and 'version' in existing_json_data then:
            # no changes were made, avoid making unnecessary change to the file modification time
            return existing_json_data
```